doc/getting-started.rst

```diff
@@ -0,0 +1,596 @@
+version for a requirement (subject to requirement constraints), it
+appends the version to the named file, along with a comment saying
+when and why the requirement is installed.  If you later want to
+upgrade a dependency, just edit this file with the new version, or to
+remove the entry altogether and Buildout will add a new entry the next
+time it runs.
+
+We also used the ``show-picked-versions`` to tell Buildout to tell us
+when it got (picked) the newest version of a requirement.
+
+When versions are pinned, Buildout doesn't look for new versions of
+the requirements, which can speed buildouts quite a bit. In fact, The
+``-N`` option doesn't provide any speedup for projects whose
+requirement versions are all pinned.
+
+When should you pin versions?
+_____________________________
+
+The rule of thumb is that you should pin versions for a whole system,
+such as an application or service.  You do this because after
+integration tests, you want to be sure that you can reproduce the
+tested configuration.
+
+You shouldn't pin versions for a component, such as a library, because
+doing so inhibits the ability for users of your component to integrate it
+with their dependencies, which may overlap with yours.  If you know
+that your component only works a range of versions of some dependency,
+the express the range in your project requirements. Don't require
+specific versions.
+
+Buildout versions and automatic upgrade
+---------------------------------------
+
+In the interest of repeatability, Buildout can upgrade itself or its
+dependencies to use the newest versions or downgrade to respect pinned
+versions.  This only happens if you run Buildout from a buildout's own
+``bin`` directory.  If you've been running the examples, you may have
+noticed the message::
+
+  Not upgrading because not running a local buildout command.
+
+We can use Buildout's ``bootstrap`` command to install a local
+buildout script::
+
+  buildout bootstrap
+
+Then, if the installed script is used::
+
+  bin/buildout
+
+Then Buildout will upgrade or downgrade to be consistent with version
+requirements.  See the :doc:`bootstrapping topic
+<topics/bootstrapping>` to learn more about bootstrapping.
+
+Python development projects
+===========================
+
+A very common Buildout use case is to manage the development of a
+library or main part of an application written in Python.  Buildout
+facilitates this with the ``develop`` option::
+
+   [buildout]
+   develop = .
+   ...
+
+The ``develop`` option takes one more more paths to project `setup.py
+<https://docs.python.org/3.6/distutils/setupscript.html>`_ files or,
+more commonly, directories containing them. Buildout then creates
+"develop eggs" [#develop-eggs]_ for the corresponding projects.
+
+With develop eggs, you can modify the sources and the modified sources
+are reflected in future Python runs (or after `reloads
+<https://docs.python.org/3/library/imp.html?highlight=reload#imp.reload>`_).
```

Comment:
The `imp` module is deprecated since Python 3.4 and tells you to use https://docs.python.org/3/library/importlib.html#importlib.reload instead.